### PR TITLE
add execute_sql API to fix ElasticSearchSQLHook

### LINKED ISSUE #41486: ElasticsearchSQLHook fails with AttributeError: __enter__
### Apache Airflow Provider(s)

elasticsearch

### Versions of Apache Airflow Providers

apache-airflow-providers-common-io==1.3.2
apache-airflow-providers-common-sql==1.14.0
apache-airflow-providers-elasticsearch==5.4.1
apache-airflow-providers-fab==1.1.1
apache-airflow-providers-ftp==3.9.1
apache-airflow-providers-http==4.11.1
apache-airflow-providers-imap==3.6.1
apache-airflow-providers-smtp==1.7.1
apache-airflow-providers-sqlite==3.8.1

### Apache Airflow version

v2.9.2

### Operating System

v2.9.2

### Deployment

Virtualenv installation

### Deployment details

_No response_

### What happened

[Documentation's usage example](https://airflow.apache.org/docs/apache-airflow-providers-elasticsearch/5.4.1/hooks/elasticsearch_sql_hook.html) for ElasticsearchSQLHook:
```
from airflow.providers.elasticsearch.hooks.elasticsearch import ElasticsearchSQLHook

es = ElasticsearchSQLHook(elasticsearch_conn_id='elasticsearch')

# Handle ES conn with context manager
with es.get_conn() as es_conn:
    tables = es_conn.execute("SHOW TABLES")
    for table, *_ in tables:
        print(f"table: {table}")
```
fails with:
```
Traceback (most recent call last):
  File "/home/port/projects/Encortex/airflow/new_test.py", line 6, in <module>
    with es.get_conn() as es_conn:
AttributeError: __enter__
```

### What you think should happen instead

_No response_

### How to reproduce

Implement [Documentation's usage example](https://airflow.apache.org/docs/apache-airflow-providers-elasticsearch/5.4.1/hooks/elasticsearch_sql_hook.html).

### Anything else

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by Owen-CH-Leung**
Fixes #41486 

The current `ElasticsearchSQLHook` is not working since it is still using `elasticsearch-dbapi` which has already been deprecated last year. This PR fixes the issue by adding a `execute_sql` API, which simply forwards the SQL query to the underlying official elasticsearch client.

Docs are also updated to show a working example of `ElasticsearchSQLHook`. A Screencap of the log output: 

image

### CODE REVIEW

**Comment by eladkal** [comment]
Static checks are failing :(

**Comment by Owen-CH-Leung** [comment]
I think the 3 failing tests are `Tests / Basic tests / Test Airflow release commands` , `Tests / Provider checks / Provider packages sdist build and install` , `Tests / Integration Tests / Integration Tests: mssql ` , which are not related to static check ?

**Review by eladkal — APPROVED** [review]

File: airflow/providers/elasticsearch/hooks/elasticsearch.py [hunk 1]
```diff
@@ -23,12 +23,15 @@
 
 from deprecated import deprecated
 from elasticsearch import Elasticsearch
+from elasticsearch.client import SqlClient
```
Comment (eladkal) [inline airflow/providers/elasticsearch/hooks/elasticsearch.py:26]: This import causes deprecation warning
https://github.com/elastic/elasticsearch-py/blob/4ee46509e236bf30f540d173b48bb7650159556d/elasticsearch/client.py#L75-L80

Comment (Owen-CH-Leung) [reply airflow/providers/elasticsearch/hooks/elasticsearch.py:26]: Indeed...We should access the SQL client directly via es instance. I'll file a PR to fix it later

**Pull request merged**